### PR TITLE
Use NFI for resolving the locale Encoding

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: src/main/java/org/truffleruby/core/encoding/EncodingManager.java [hunk 1]
```diff
@@ -46,13 +52,38 @@
...
+        final int codeset = LangInfo.CODESET.intValue();
+
+        // char *nl_langinfo(nl_item item);
+        // nl_item is int on at least Linux, macOS & Solaris
+        final TruffleNFIPlatform nfi = nativePlatform.getTruffleNFI();
+        final NativeFunction nl_langinfo = nfi.getFunction("nl_langinfo", 1, "(sint32):string");
+
+        final long address = nfi.asPointer((TruffleObject) nl_langinfo.call(codeset));
+        final byte[] bytes = new Pointer(address).readZeroTerminatedByteArray(0);
+        final String localeEncodingName = new String(bytes, StandardCharsets.ISO_8859_1);
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/core/encoding/EncodingManager.java]: This one's tricky. The charset should be the locale encoding's charset, but we don't know what that is yet.

Comment (eregon) [reply src/main/java/org/truffleruby/core/encoding/EncodingManager.java]: Yeah, it's dependent on itself :)
I think a sane locale setup implies the locale name should be only 7-bit characters.
If it's not, we will just fallback to US-ASCII and consider it an invalid locale with ISO_8859_1.

Comment (nirvdrum) [reply src/main/java/org/truffleruby/core/encoding/EncodingManager.java]: Yeah. MRI treats all encoding names as US-ASCII. We do the same now.

**Comment by eregon** [comment]
@nirvdrum I added a commit to clarify this ordering.

**Pull request merged**